packages/playwright-test/src/matchers/toMatchSnapshot.ts

```diff
@@ -232,6 +241,86 @@ export function toMatchSnapshot(
   return helper.handleDifferent(received, expected, result.diff, result.errorMessage, undefined);
 }
 
+export async function toHaveScreenshot(
+  this: ReturnType<Expect['getState']>,
+  pageOrLocator: Page | Locator,
+  nameOrOptions: NameOrSegments | { name?: NameOrSegments } & HaveScreenshotOptions = {},
+  optOptions: HaveScreenshotOptions = {}
+): Promise<SyncExpectationResult> {
+  const testInfo = currentTestInfo();
+  if (!testInfo)
+    throw new Error(`toHaveScreenshot() must be called during the test`);
+  const helper = new SnapshotHelper(testInfo, 'png', nameOrOptions, optOptions);
+  const [page, locator] = pageOrLocator.constructor.name === 'Page' ? [(pageOrLocator as PageEx), undefined] : [(pageOrLocator as Locator).page() as PageEx, pageOrLocator as Locator];
+
+  if (this.isNot) {
+    if (!fs.existsSync(helper.snapshotPath))
+      return helper.handleMissingNegated();
+
+    // Having `errorMessage` means we timed out while waiting
+    // for screenshots not to match, so screnshots
+    // are actually the same in the end.
+    const isDifferent = !(await page._expectScreenshot({
+      expected: await fs.promises.readFile(helper.snapshotPath),
+      isNot: true,
+      locator,
+      comparatorOptions: helper.comparatorOptions,
+      timeout: currentExpectTimeout(helper.comparatorOptions),
+    })).errorMessage;
+    return isDifferent ? helper.handleDifferentNegated() : helper.handleMatchingNegated();
+  }
+
+  if (helper.updateSnapshots === 'all' || !fs.existsSync(helper.snapshotPath)) {
```

Comment:
Let's not enter this block when `updateSnapshots === 'none'` and there is no snapshot.